### DISCUSSION TITLE
[POQ] POQ Item Location Type and POQ Item Location representation in swagger file

### DISCUSSION THREAD

**Discussion by mariusz-amartus** [MEF-GIT/MEF-LSO-Sonata-SDK#172] [Team Posts]
Hi,

I received following question with regards to the LSO Sonata SDK R3:

> 1. The swagger does not match with the spec (MEF79/80)
> For example, in MEF79 section 9.4.1.1 Buyer Specification of POQ Items, there are POQ Item Location Type and POQ Item Location
> ![poq_item](https://user-images.githubusercontent.com/52410146/66311546-02aa3080-e90f-11e9-8846-77f8ff3c6432.png)
> In swagger, there are no such fields available, but only an attribute called place:
> ![swagger_type](https://user-images.githubusercontent.com/52410146/66311588-1c4b7800-e90f-11e9-992f-211c7c3f2fd3.png)

Here's my understanding for the above:

> Attribute "POQ Item Location Type" should be provided with "product (Product) -> place (RelatedPlaceReforValue) -> type". And depending on the "type" value (FormattedAddress, FieldedAddress, GeographicLocation or ReferencedAddress) the attributes specific for the selected type should be added to "ProductOfferingQualificationItem" to provide "POQ Item Location".

@bigludo7, @fahimsabir-colt, others,
Could you please share your expertise on this? I think the answer could be useful for all potenital implementors of LSO Sonata APIs.

Thanks,
Mariusz
LSO SDK Community Manager

**Comment by bigludo7** [comment]
Hello,

Some elements:
The @type attribute in the RelatedPlaceReforValue is tagged as a discriminator.

Then FieldedAddress, FormattedAddress, ReferencedAddress and GeographicLocation are defined in the swagger and has a 
    allOf:
    - $ref: "#/definitions/RelatedPlaceReforValue"
These class are extending the placeRefOrValue. Disciminator is use to point to the used strucutre.

Hope it helps
Ludovic
